tests/codegen/binop_int_promotion.cpp

```diff
@@ -0,0 +1,14 @@
+#include "common.h"
+
+namespace bpftrace {
+namespace test {
+namespace codegen {
+
+TEST(codegen, binop_int_promotion)
+{
+  test("kretprobe:f { $x = (uint32)5; $x += (uint32)1 }", NAME);
```

Comment:
Technically speaking, there's no promotion here. What about doing
```suggestion
  test("kretprobe:f { $x = (uint32)5; $x += (uint16)1 }", NAME);
```
to check that the upcast is correctly generated?

Reply:
Yeah good point. I think I just named it poorly - idea was to test the regression. But this catches more.